src/cmd/linuxkit/pkglib/build.go

```diff
@@ -109,31 +156,44 @@ func (p Pkg) Build(bos ...BuildOpt) error {
 		}
 	}
 
+	writer := bo.writer
+	if writer == nil {
+		writer = os.Stdout
+	}
+
 	arch := runtime.GOARCH
+	ref, err := reference.Parse(p.Tag())
+	if err != nil {
+		return fmt.Errorf("could not resolve references for image %s: %v", p.Tag(), err)
+	}
 
-	if !p.archSupported(arch) {
-		fmt.Printf("Arch %s not supported by this package, skipping build.\n", arch)
-		return nil
+	for _, platform := range bo.platforms {
+		if !p.archSupported(platform.Architecture) {
+			return fmt.Errorf("Arch %s not supported by this package, skipping build.\n", platform.Architecture)
```

Comment:
`./pkglib/build.go:172:22: error strings should not be capitalized or end with punctuation or a newline`